### PR TITLE
fix(go): add GOPROXY username/password to global secrets list

### PR DESCRIPTION
**Pull request by rarkins**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/go/index.ts [hunk 1]
```diff
@@ -73,3 +76,14 @@ export class GoDatasource extends Datasource {
...
+// istanbul ignore if
+if (is.string(process.env.GOPROXY)) {
+  const uri = parseUrl(process.env.GOPROXY);
+  if (uri.username) {
+    addSecretForSanitizing(uri.username, 'global');
+  }
+  if (uri.password) {
+    addSecretForSanitizing(uri.password, 'global');
+  }
+}
```
Comment (viceice) [inline lib/modules/datasource/go/index.ts:89]: Shouldn't we do this in `config/parse/env`? 🤔

Comment (rarkins) [reply lib/modules/datasource/go/index.ts:89]: We could.. I was hoping to keep datasource-specific stuff out of `workers/` though. Also we don't really parse it back into config like the rest of config/parse/env.

**Comment by viceice** [comment]
Needs rebase

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

File: lib/modules/datasource/go/index.ts [hunk 2]
```diff
@@ -73,3 +76,14 @@ export class GoDatasource extends Datasource {
...
   }
 }
+
+// istanbul ignore if
+if (is.string(process.env.GOPROXY)) {
+  const uri = parseUrl(process.env.GOPROXY);
+  if (uri.username) {
+    addSecretForSanitizing(uri.username, 'global');
+  }
+  if (uri.password) {
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/modules/datasource/go/index.ts]: It's not part of URL as far as I see

**Review by viceice — APPROVED** [review]

**Pull request merged**